### PR TITLE
Always diagnose request evaluator cycles

### PR DESCRIPTION
**Pull request by slavapestov**
Turns out only minor fixes were needed, so let's enable this and make sure we don't regress.

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

**Comment by slavapestov** [comment]
@swift-ci Please test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

**Comment by slavapestov** [comment]
@swift-ci Please test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

**Review by DougGregor — COMMENTED** [review]
Very excited for this!

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

File: lib/Sema/TypeCheckDeclObjC.cpp [hunk 1]
```diff
@@ -1886,8 +1890,7 @@ bool swift::fixDeclarationObjCName(InFlightDiagnostic &diag, ValueDecl *decl,
     return false;
   }
 
-  // Determine the Objective-C name of the declaration.
-  ObjCSelector name = *decl->getObjCRuntimeName();
+  auto name = *nameOpt;
   auto targetName = *targetNameOpt;
```
Comment (brentdax) [inline lib/Sema/TypeCheckDeclObjC.cpp]: I was a little confused by these two lines when I read them. I guess the idea is that these should only be `None` when the declaration is implicit or a subscript? If so, maybe there should be a comment here or a sentence in the doc comment to that effect.

**Pull request merged**